### PR TITLE
refactor: port components model

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- port `Components` model
- add unit tests

**Related issue(s)**
Part of #481 
Part of #482 

cc @smoya

### CODE REVIEW

File: src/models/components.ts [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
+  operations(): Record<string, OperationInterface>;
+  messages(): Record<string, MessageInterface>;
+  schemas(): Record<string, SchemaInterface>;
+  channelParameters(): Record<string, ChannelParameterInterface>;
+  serverVariables(): Record<string, ServerVariableInterface>;
+  operationTraits(): Record<string, OperationTraitInterface>;
+  messageTraits(): Record<string, MessageTraitInterface>;
+  correlationIds(): Record<string, CorrelationIdInterface>;
+  externalDocs(): Record<string, ExternalDocumentationInterface>;
+  tags(): Record<string, TagInterface>;
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/models/components.ts:30]: Why are you adding components that have yet to be accepted into the spec? 🤔

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/models/components.ts:30]: Why not? Operations in 3.0.0 will land in `components` anyway.

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/models/components.ts:30]: Cause if the changes are rejected, they should not already be part of the parser 🙂 You still follow RFC.

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/models/components.ts]: Ok, removed :P

**Review by jonaslagoni — APPROVED** [review]

**Comment by magicmatatjahu** [comment]
@jonaslagoni Thanks!

**Pull request merged**